src/streaming/rules/baseUrlResolution/DVBSelector.js

```diff
@@ -0,0 +1,147 @@
+ *  other materials provided with the distribution.
+ *  * Neither the name of Dash Industry Forum nor the names of its
+ *  contributors may be used to endorse or promote products derived from this software
+ *  without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS AS IS AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ *  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ *  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+import FactoryMaker from '../../../core/FactoryMaker.js';
+
+function DVBSelector(config) {
+
+    let instance;
+
+    const blacklistController = config.blacklistController;
+
+    function getNonBlacklistedBaseUrls(urls) {
+        var removedPriorities = [];
+
+        const samePrioritiesFilter = function (el) {
+            if (removedPriorities.length) {
+                if ((el.dvb_priority) &&
+                        (removedPriorities.indexOf(el.dvb_priority) !== -1)) {
+                    return false;
+                }
+            }
+
+            return true;
+        };
+
+        const serviceLocationFilter = function (baseUrl) {
+            if (blacklistController.contains(baseUrl.serviceLocation)) {
+                // whenever a BaseURL is removed from the available list of
+                // BaseURLs, any other BaseURL with the same @priority
+                // value as the BaseURL being removed shall also be removed
+                if (baseUrl.dvb_priority) {
+                    removedPriorities.push(baseUrl.dvb_priority);
+                }
+
+                // all URLs in the list which have a @serviceLocation
+                // attribute matching an entry in the blacklist shall be
+                // removed from the available list of BaseURLs
+                return false;
+            }
+
+            return true;
+        };
+
+        return urls.filter(serviceLocationFilter).filter(samePrioritiesFilter);
+    }
+
+    function selectByWeight(availableUrls) {
+        const prioritySorter = function (a, b) {
+            var diff = a.dvb_priority - b.dvb_priority;
+            return isNaN(diff) ? 0 : diff;
+        };
+
+        const topPriorityFilter = function (baseUrl, idx, arr) {
+            return !idx || (
+                (arr[0].dvb_priority && baseUrl.dvb_priority) &&
```

Comment:
Yes, I can see this could be confusing but it is definitely correct - I should probably have put a description in.

For everyone else: by the time this filter is applied, the list has already been sorted into descending priority order, so we know that the first entry in the list must have the highest priority. The first is automatically selected, then the remaining entries are then selected only if their priority matches the first, yielding a list of all entries with the highest priority.